sched/semaphore/spinlock.c

```diff
@@ -137,32 +155,52 @@ void spin_lock_wo_note(FAR volatile spinlock_t *lock)
  *
  ****************************************************************************/
 
-spinlock_t spin_trylock(FAR volatile spinlock_t *lock)
+bool spin_trylock(FAR volatile spinlock_t *lock)
 {
 #ifdef CONFIG_SCHED_INSTRUMENTATION_SPINLOCKS
   /* Notify that we are waiting for a spinlock */
 
   sched_note_spinlock(this_task(), lock, NOTE_SPINLOCK_LOCK);
 #endif
 
-  if (up_testset(lock) == SP_LOCKED)
+#ifdef CONFIG_TICKET_SPINLOCK
+  uint16_t ticket = atomic_load(&lock->tickets.next);
+
+  spinlock_t old =
+    {
+    {
+        ticket, ticket
+    }
+    };
+
+  spinlock_t new =
+    {
+    {
+        ticket, ticket + 1
+    }
+    };
+
+  if (atomic_compare_exchange_strong(&lock->value, &old.value, new.value))
+#else /* CONFIG_TICKET_SPINLOCK */
+  if (up_testset(lock) == SP_UNLOCKED)
+#endif /* CONFIG_TICKET_SPINLOCK */ 
     {
 #ifdef CONFIG_SCHED_INSTRUMENTATION_SPINLOCKS
       /* Notify that we abort for a spinlock */
 
       sched_note_spinlock(this_task(), lock, NOTE_SPINLOCK_ABORT);
 #endif
-      SP_DSB();
-      return SP_LOCKED;
+      SP_DMB();
```

Comment:
why change

Reply:
After thinking, the change isn't necessary.
revert.